Sources/MapboxMaps/Foundation/MapView.swift

```diff
@@ -174,30 +173,90 @@ open class MapView: UIView {
-        dependencyProvider = MapViewDependencyProvider()
-        notificationCenter = dependencyProvider.makeNotificationCenter()
-        bundle = dependencyProvider.makeBundle()
-        super.init(frame: frame)
-        commonInit(mapInitOptions: mapInitOptions, overridingStyleURI: nil)
+    @available(iOSApplicationExtension, unavailable)
+    public convenience init(frame: CGRect, mapInitOptions: MapInitOptions = MapInitOptions()) {
+        let orientationProvider: InterfaceOrientationProvider
+        if #available(iOS 13, *) {
+            orientationProvider = DefaultInterfaceOrientationProvider()
+        } else {
+            orientationProvider = UIApplicationInterfaceOrientationProvider()
+        }
+        self.init(frame: frame,
+                  mapInitOptions: mapInitOptions,
+                  dependencyProvider: MapViewDependencyProvider(),
+                  orientationProvider: orientationProvider,
+                  urlOpener: DefaultAttributionURLOpener())
     }
 
+    /// Initialize a MapView
+    /// - Parameters:
+    ///   - frame: frame for the MapView.
+    ///   - mapInitOptions: `MapInitOptions`; default uses
+    ///    `ResourceOptionsManager.default` to retrieve a shared default resource option, including the access token.
+    ///   - orientationProvider: User interface orientation provider
+    ///   - urlOpener: Attribution URL opener
+    @available(iOS, deprecated: 13, message: "Use init(frame:mapInitOptions:urlOpener:) instead")
+    public convenience init(frame: CGRect,
+                            mapInitOptions: MapInitOptions = MapInitOptions(),
+                            orientationProvider: InterfaceOrientationProvider,
+                            urlOpener: AttributionURLOpener) {
+        self.init(frame: frame,
+                  mapInitOptions: mapInitOptions,
+                  dependencyProvider: MapViewDependencyProvider(),
+                  orientationProvider: orientationProvider,
+                  urlOpener: urlOpener)
+    }
+
+    /// Initialize a MapView
+    /// - Parameters:
+    ///   - frame: frame for the MapView.
+    ///   - mapInitOptions: `MapInitOptions`; default uses
+    ///    `ResourceOptionsManager.default` to retrieve a shared default resource option, including the access token.
+    ///   - urlOpener: Attribution URL opener
+    @available(iOS 13.0, *)
+    public convenience init(frame: CGRect,
+                            mapInitOptions: MapInitOptions = MapInitOptions(),
+                            urlOpener: AttributionURLOpener) {
+        self.init(frame: frame,
+                  mapInitOptions: mapInitOptions,
+                  dependencyProvider: MapViewDependencyProvider(),
+                  orientationProvider: DefaultInterfaceOrientationProvider(),
+                  urlOpener: urlOpener)
+    }
+
+    @available(iOSApplicationExtension, unavailable)
     required public init?(coder: NSCoder) {
```

Comment:
Can you use xibs or storyboards together with App Extensions? If so, what would happen if you tried to do so with an instance of MapView?